src/modules/home/components/single-property/single-property-amenity.tsx

```diff
@@ -0,0 +1,142 @@
+import { useState, useMemo } from 'react';
+import Avatar from '@mui/material/Avatar';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Divider from '@mui/material/Divider';
+import Button from '@mui/material/Button';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import dayjs, { type Dayjs } from 'dayjs';
+import { SinglePropertyAmenityStyles } from './single-property-amenity.styles';
+import { type AmenityInterface } from '../../../../types/amenity.types';
+
+interface SinglePropertyAmenityProps {
+  owner: string;
+  amenities: AmenityInterface[];
+  description: string;
+  pricePerNight: number;
+}
+
+function SinglePropertyAmenity({ owner, amenities, description, pricePerNight }: SinglePropertyAmenityProps) {
+  const [showAll, setShowAll] = useState(false);
+  const [checkInDate, setCheckInDate] = useState<Dayjs | null>(null);
+  const [checkOutDate, setCheckOutDate] = useState<Dayjs | null>(null);
+
+  const handleShowAll = () => {
+    setShowAll(!showAll);
+  };
+
+  const displayedAmenities = showAll ? amenities : amenities.slice(0, 10);
+
+  const calculatePrice = useMemo(() => {
+    if (
+      checkInDate &&
+      checkOutDate &&
+      checkOutDate.isValid() &&
+      checkInDate.isValid() &&
+      checkOutDate.diff(checkInDate, 'day') >= 0
+    ) {
+      const diffInDays = checkOutDate.diff(checkInDate, 'day');
+      const basePrice = (diffInDays === 0 ? 1 : diffInDays) * pricePerNight;
+      const serviceFee = 0.1 * basePrice;
+      return { basePrice, serviceFee, diffInDays };
```

Comment:
Rename the serviceFee to something more descriptive, like withServiceFee or overallPrice. Because it's holding the overall cost of stay, if I'm not mistaken.